cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Node.kt

```diff
@@ -286,6 +286,10 @@ abstract class Node() :
         Overlays(this, mirrorProperty = OverlayNode::underlyingNodeEdge, outgoing = true)
     var overlays by unwrapping(Node::overlayEdges)
 
+    override fun collectAssumptions(): Set<Assumption> {
+        return super.collectAssumptions() + (component?.assumptions ?: emptySet())
```

Comment:
Should it be `assumptions` or `collectAssumptions`?

Reply:
I had collectAssumptions before, which creates a loop. Here I knew that the component would be a simple node and `.assumptions` would be enough. Else, I would have to override the ´collectAssumptions´ specifically for `Component`s to not recursively call `collectAssumptions` because a components component is itself.

Reply:
I'd suggest to override the implementation for `Component` in this case because this system sounds very error-prone if the user just has to known this or will create loops or not receive any meaningful result.